src/main/java/fi/vm/yti/datamodel/api/v2/endpoint/Schema.java

```diff
@@ -41,126 +47,128 @@
+		Model metadataModel = jenaService.getSchema(pid);
+		SchemaInfoDTO schemaDTO = mapper.mapToSchemaDTO(pid, metadataModel);
+		check(authorizationManager.hasRightToAnyOrganization(Set.of(schemaDTO.getOrganization())));
 
-    
-    @Operation(summary = "Get a schema metadata")
-    @ApiResponse(responseCode = "200", description = "")
-    @GetMapping(value = "/schema/{pid}", produces = APPLICATION_JSON_VALUE)
-    public SchemaInfoDTO getSchemaMetadata(@PathVariable String pid){
-    	var jenaModel = jenaService.getSchema(pid);
-    	return mapper.mapToSchemaDTO(pid, jenaModel);
-    }
-    
-    @Operation(summary = "Upload and associate a schema description file to an existing schema")
-    @ApiResponse(responseCode = "200", description = "")
-    @SecurityRequirement(name ="Bearer Authentication")
-    @PutMapping(path="/schema/{pid}/upload", produces = APPLICATION_JSON_VALUE, consumes = "multipart/form-data")
-    public SchemaInfoDTO uploadSchemaFile(@PathVariable String pid, @RequestParam("contentType") String contentType, @RequestParam("file") MultipartFile file) {
-    	var metadataModel = jenaService.getSchema(pid);
-    	SchemaInfoDTO schemaDTO = mapper.mapToSchemaDTO(pid, metadataModel);    	
-		check(authorizationManager.hasRightToAnyOrganization(Set.of(schemaDTO.getOrganization())));		
-    	
-    	try {
-    		// transform to the internal format
-    		if(schemaDTO.getFormat() != SchemaFormat.JSONSCHEMA) {
-    			throw new RuntimeException("Unsupported schema description format");
-    		}
-    		Model schemaModel = schemaService.transformJSONSchemaToInternal(pid, file.getBytes());
-    		schemaModel.add(metadataModel);
-    		jenaService.updateSchema(pid, schemaModel);
-    		// store the original file    		
-    		storageService.storeSchemaFile(pid, contentType, file.getBytes());
-    		
-    		
-    	}catch(Exception ex) {
-    		throw new RuntimeException("Error occured while ingesting file based schema description", ex);
-    	}
-    	return mapper.mapToSchemaDTO(pid, metadataModel);
-    }
-    
-    
-    @Operation(summary = "Get original file version of the schema (if available)", description = "If the result is only one file it is returned as is, but if the content includes multiple files they a returned as a zip file.")
-    @ApiResponse(responseCode = "200", description = "")
-    @GetMapping(path = "/schema/{pid}/original")
-    public ResponseEntity<byte[]> exportOriginalFile(@PathVariable String pid) {
-    	List<StoredFile> files = storageService.retrieveAllSchemaFiles(pid);
-    	if(files.size() == 1) {
-    		StoredFile file = files.get(0);
+		try {
+			byte[] fileInBytes = file.getBytes();
+			if (schemaDTO.getFormat() == SchemaFormat.JSONSCHEMA) {
+				ValidationRecord validationRecord = JSONValidationService.validateJSONSchema(fileInBytes);
+
+				boolean validationStatus = validationRecord.isValid();
+				List<String> validationMessages = validationRecord.validationOutput();
+
+				if (validationStatus) {
```

Comment:
Rename validatationStatus variable. It is difficult to understand whether the if block should be executed or not just by reading the line "if(validatationStatus)".  Compared to for example "if(isValidJSON)"